stactools_threedep/stactools/threedep/metadata.py

```diff
@@ -0,0 +1,218 @@
+        )
+        parts = tiff_href.split('/')[-4:]
+        self.product = parts[0]
+        self.id = parts[2]
+
+    @property
+    def stac_id(self) -> str:
+        """Returns the STAC ID of this metadata.
+
+        This is the id plus the product, e.g. if the filename of the tif is
+        "USGS_1_n40w105.tif", then the STAC id is "n40w105-1".
+        """
+        return "{}-{}".format(self.id, self.product)
+
+    @property
+    def geometry(self) -> dict:
+        """Returns this item's geometry in WGS84."""
+        original_bbox = [
+            float(self.minx),
+            float(self.miny),
+            float(self.maxx),
+            float(self.maxy)
+        ]
+        return reproject_geom(THREEDEP_CRS, "EPSG:4326",
+                              mapping(box(*original_bbox)))
+
+    @property
+    def datetime(self) -> Union[datetime.datetime, None]:
+        """Returns the collection publication datetime."""
+        if self.current == "publication date":
+            return _format_date(self.pubdate)
+        else:
+            raise NotImplementedError
+
+    @property
+    def start_datetime(self) -> Union[datetime.datetime, None]:
+        """Returns the start datetime for this record.
+
+        This can be a while ago, since the national elevation dataset was
+        originally derived from direct survey data.
+        """
+        return _format_date(self.begdate)
+
+    @property
+    def end_datetime(self) -> Union[datetime.datetime, None]:
+        """Returns the end datetime for this record."""
+        return _format_date(self.enddate, end_of_year=True)
+
+    @property
+    def gsd(self) -> float:
+        """Returns the nominal ground sample distance from these metadata."""
+        if self.product == "1":
+            return 30
+        elif self.product == "13":
+            return 10
+        else:
+            raise NotImplementedError
+
+    def data_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the data asset (aka the tiff file)."""
+        return Asset(href=self._asset_href_with_extension(base, "tif"),
+                     title=self.title,
+                     description=self.description,
+                     media_type=MediaType.COG,
+                     roles=["data"])
+
+    def metadata_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the data asset (aka the tiff file)."""
+        return Asset(href=self._asset_href_with_extension(base, "xml"),
+                     media_type=MediaType.XML,
+                     roles=["metadata"])
+
+    def thumbnail_asset(self, base: str = DEFAULT_BASE) -> Asset:
+        """Returns the numbnail asset."""
+        return Asset(href=self._asset_href_with_extension(base, "jpg"),
+                     media_type=MediaType.JPEG,
+                     roles=["thumbnail"])
+
+    def derived_from_link(self, base: str = DEFAULT_BASE) -> Link:
```

Comment:
The intention of the `derived_from` link was that it allows for tracing provenance through STAC Items. In other words, the target of `derived_from` should be another STAC Item.
To point back to the original metadata that was used to create the STAC Item, use `via`:
https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#using-relation-types